superset/assets/visualizations/nvd3_vis.js

```diff
@@ -317,45 +316,33 @@ function nvd3Vis(slice, payload) {
     if (fd.x_log_scale) {
       chart.xScale(d3.scale.log());
     }
-    let xAxisFormatter;
-    if (vizType === 'bubble') {
-      xAxisFormatter = d3.format('.3s');
-    } else if (fd.x_axis_format === 'smart_date') {
-      xAxisFormatter = formatDate;
-      chart.xAxis.tickFormat(xAxisFormatter);
-    } else if (fd.x_axis_format !== undefined) {
-      xAxisFormatter = timeFormatFactory(fd.x_axis_format);
-      chart.xAxis.tickFormat(xAxisFormatter);
-    }
 
     const isTimeSeries = timeStampFormats.indexOf(fd.x_axis_format) > -1;
     // if x axis format is a date format, rotate label 90 degrees
     if (isTimeSeries) {
       chart.xAxis.rotateLabels(45);
     }
 
+    let xAxisFormatter = d3FormatPreset(fd.x_axis_format);
+    if (isTimeSeries && fd.x_axis_format) {
+      xAxisFormatter = d3TimeFormatPreset(fd.x_axis_format);
+    }
     if (chart.hasOwnProperty('x2Axis')) {
       chart.x2Axis.tickFormat(xAxisFormatter);
       height += 30;
     }
+    if (chart.hasOwnProperty('xAxis')) {
```

Comment:
we should check that `chart.axis` is defined and has the key `tickFormat`.  can we assume that if they key is present that it has a value?